### PR TITLE
Human readable uid

### PR DESCRIPTION
**Pull request by ADimeo**
This branch implements a UI window which allows users to set their name for collaboration.

It does not deal with anything network related, and does not have full test coverage, see #6 for details regarding that.

### CODE REVIEW

**Comment by arminWel** [comment]
I know, that there may not be foll test coverage due to #6. Nevertheless all test, which are already implemented should work in a PR. Is there a specific reason for them not to in this case?

File: packages/CollabCodeEdit-Client.package/CCEAccount.class/class/default.st [hunk 1]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEAccount.class/class/default.st]: Always classify methods.

File: packages/CollabCodeEdit-Client.package/CCEAccount.class/class/reset.st [hunk 2]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEAccount.class/class/reset.st]: Always classify methods.

File: packages/CollabCodeEdit-Client.package/CCEAccount.class/instance/humanReadableName.st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+accessing
+humanReadableName
+	^humanReadableName ifNil: [
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEAccount.class/instance/humanReadableName.st]:
```suggestion
	^ humanReadableName ifNil: [ 
```

File: packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/setUp.st [hunk 4]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/setUp.st]: Always classify methods.

File: packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testChangeHumanReadableName.st [hunk 5]
```diff
@@ -0,0 +1,7 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testChangeHumanReadableName.st]: Always classify methods.

File: packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testChangeHumanReadableName.st [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+testChangeHumanReadableName
+	| everChangingName |
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testChangeHumanReadableName.st:3]:
```suggestion
	| everChangingName |
	
```

File: packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testHumanReadableNameDefaultsToAuthorInitials.st [hunk 7]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEAcccountTests.class/instance/testHumanReadableNameDefaultsToAuthorInitials.st]: Always classify methods.

File: packages/CollabCodeEdit-Tests.package/CCEModelTests.class/instance/testBuildGlobalStateAccountDisplayWithCCEAccountAsModel.st [hunk 8]
```diff
@@ -0,0 +1,7 @@
+tests-building
+testBuildGlobalStateAccountDisplayWithCCEAccountAsModel
+	| spec |
+	spec := model buildGlobalStateAccountDisplayWith: toolbuilder.
+	self assert: (spec isKindOf: ToolBuilderSpec).
+	self assert: (spec model == CCEAccount default)
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEModelTests.class/instance/testBuildGlobalStateAccountDisplayWithCCEAccountAsModel.st]:
```suggestion
```

File: packages/CollabCodeEdit-Tests.package/CCEModelTests.class/instance/testBuildGlobalStateDisplayReturnsCollection.st [hunk 9]
```diff
@@ -0,0 +1,7 @@
+tests-building
+testBuildGlobalStateDisplayReturnsCollection
+	| spec |
+	spec := model buildGlobalStateDisplayWith: toolbuilder.
+	self assert: (spec isKindOf: Collection).
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEModelTests.class/instance/testBuildGlobalStateDisplayReturnsCollection.st]:
```suggestion
	self assert: (spec isKindOf: Collection)
```
Delete the trailing whitespace

**Review by arminWel — CHANGES_REQUESTED** [review]

File: packages/CollabCodeEdit-Client.package/CCEModel.class/instance/buildGlobalStateAccountDisplayWith..st [hunk 10]
```diff
@@ -0,0 +1,12 @@
+building
+buildGlobalStateAccountDisplayWith: aBuilder
+	^ aBuilder pluggableTextSpec new 
+		model: CCEAccount default;
```
Comment (simonstadlinger) [inline packages/CollabCodeEdit-Client.package/CCEModel.class/instance/buildGlobalStateAccountDisplayWith..st]: I would suggest to handle this as all our other singletons: create a instance variable in CCEModel and use this one for all usages of this singleton.
```suggestion
		model: self imageAccount;
```

Comment (simonstadlinger) [reply packages/CollabCodeEdit-Client.package/CCEModel.class/instance/buildGlobalStateAccountDisplayWith..st]: I know, in the existing code base at your state, this is not how we deal with singletons. in SessionManagment we do.

**Review by simonstadlinger — CHANGES_REQUESTED** [review]

**Review by simonstadlinger — APPROVED** [review]
well done!

**Review by simonstadlinger — APPROVED** [review]

**Pull request merged**